Reorganize FMCDAQ3, ADRV9361_BOB, SYNCHRONA

trupples opened the pull request.

trupples committed:
Fix typo in test setup script

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Add FMCDAQ3 production tests

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Export BOARD_SERIAL to sub-shells

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Use proper pytest file for ADRV9361_BOB

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Set proper executable permissions to .sh files

Scripts which are directly run (e.g. production.sh) should be executable.
Scripts which are always `source`-d (e.g. lib/*) should not be.

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Add libzstd1 dependency

Required in the build step of libiio

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Add dpkg lock timeout to apt install

setup_env.sh sometimes fails because apt can't get the dpkg lock because of other running apt processes, such as You need to be root to run this application. This change allows for a 60s grace period before failing

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Add datetime synchronization setup step

On a fresh kuiper prebuilt image, the current date&time are not synchronized to anything and are usually long in the past. This breaks apt install and potentially other utilities as well. This setup step tries to first use NTP (which I found is not available on said fresh kuiper image) and alternatively queries a well-known HTTP server (in this case, google.com) for the current date&time, as a low precision but still OK fallback.

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Return a zero status on errors in ADRV9361_BOB

trupples committed:
Add genalyzer to setup steps

To make this work, the following commit was also squashed-in:

Prepend to LD_LIBRARY_PATH instead of overwriting

The current implementation traces back to fed5304 from 5 years ago, but
combined with `setup_env.sh` adding `vars.sh` to be run in `.bashrc`,
this overwrites any other directories in LD_LIBRARY_PATH, sometimes
breaking expected `ld.so` behaviour.

trupples committed:
Install latest pip and setuptools before tests

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Don't disable pi wifi

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Remove common_build_tool + copy m1k download logic

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

constmonica commented on main_tests/adrv9361_bob/rf_test.sh:
```diff
@@ -3,6 +3,6 @@ ANSWER=1
 
 echo
-python3 -m pytest --color yes -vs $SCRIPT_DIR/../work/pyadi-iio/test/test_fmcomms2-3_prod.py --uri="ip:analog.local" --adi-hw-map --hw=adrv9361
+python3 -m pytest --color yes -vs $SCRIPT_DIR/../work/pyadi-iio/test/test_adrv9361_prod.py --uri="ip:analog.local" --adi-hw-map --hw=adrv9361
```
does this test_adrv9361_prod.py file actually exist?

trupples replied:
It was added in the fmcomms_scpi branch, commit [476bbcc](https://github.com/analogdevicesinc/pyadi-iio/commit/476bbccecd42616f6cb6217bc9cf118e18e71b42).

constmonica commented on main_tests/fmcdaq3/production.sh:
```diff
@@ -0,0 +1,19 @@
+MODE="$1"
+case $MODE in
+	"FMCDAQ3 Test")
+		"$SCRIPT_DIR/rf_test.sh";
```
why the quotation marks?

trupples replied:
I added the quotes to allow for spaces in the file path. Not absolutely necessary for the prodtest images - this is likely a harmless leftover from one of my test runs. Should I remove it the quotes consistency's sake?

constmonica replied:
yes

ranechita commented on main_tests/production_FMCDAQ3.sh:
```diff
@@ -0,0 +1,38 @@
+				wait_for_board_online
+				export BOARD_SERIAL=$(get_board_serial)
+				echo_blue "Starting FMCDAQ3 Test"
+				production "crr" "$opt" "FMCDAQ3"
```
it's outside the scope of this pull request, but for future reference I think the "crr"/target parameter is redundant, i think only talise uses it for something minor. Perhaps it should be removed in future versions

ranechita commented on main_tests/setup_env.sh:
```diff
@@ -391,6 +442,16 @@ dhcp-range=192.168.0.100,192.168.0.150,24h
+	sudo timedatectl set-ntp true && return
+
+	# Alternatively, use a much less precise source, but still good enough for certificates to be valid
+	sudo date +"%d %b %Y %T %Z" -s "$(curl -s --head http://google.com | grep '^Date:' | cut -d' ' -f 3-)"
```
just a nitpick but since this step and prereqs install require internet, maybe it would be useful to add a short connection check in the beginning of the script, to avoid going further if the device is not connected to the internet (maybe just a ping?)

trupples replied:
Does [def3e61](https://github.com/analogdevicesinc/production-tests/pull/20/commits/def3e6164623361f59a4c4dc7ed0f894984bc6c8) look alright?

ranechita replied:
looks good to me

trupples committed:
Add check_internet step in setup_env.sh

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Code consistency tweaks

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Reorganize synchrona tests

This is a squashed commit which introduces the following:

1. Move synchrona test files to the proper folder (synch -> synchrona)
2. Fix bash syntax errors in lib/utils.sh (whitespace around env var assignments)
3. Change DUT subnet to /16 instead of /24
4. Copy the test files to the DUT using rsync at the beginning of each test
5. Detect ADT7422 hwmon index instead of hardcoding hwmon1 (hwmon indices may change, they are not bound to a device)

trupples committed:
Merge branch 'reorganize_tests' into trupples/merge_tidy

ranechita reviewed (approved):
take it out of draft before merging, otherwise looks good to me

trupples commented:
Closed as very stale. Will restructure into multiple, tidier, PRs.
